### PR TITLE
dev-libs/apache-arrow: Fix all the `apache-arrow-9.0.0-r1` badness

### PR DESCRIPTION
**Pull request by leycec**
Greetings, fabulous Apache packagers! This PR resolves *all* of the Apache Arrow 9.0.0 badness reported in issue #21 – including:

* Temporarily disabling the `lz4` USE flag by forcing `-DARROW_WITH_LZ4=OFF` in the `src_configure()` phase. Doing so resolves this CMake configuration error: `The dependency target "lz4::lz4" of target "arrow_dependencies" does not exist.`
* Bumping `EAPI` and `PYTHON_COMPAT` to correspond to those of the 4.0.1 ebuild.
* Adding unstable ARM architectures to `KEYWORDS`, which Arrow 9.0.0 supports... *in theory.*
* Refactoring `DEPEND` and `RDEPEND` to mostly correspond to those of the 4.0.1 ebuild.
* Removing the inapplicable `BDEPEND`.
* Adding the `pkg_setup()` phase from the 4.0.1 ebuild.
* Removing the obsolete `apache-arrow-9.0.0` and `apache-arrow-9.0.0-r1` ebuilds.

Phew! Rebuilding both Arrow and PyArrow against these changes totally resolved a [related Streamlit issue](https://github.com/streamlit/streamlit/issues/5683) I stumbled over earlier this week. I am grateful and will now sleep for the entire weekend. :sleeping: 

Thanks so much for maintaining this amazing overlay, everybody.

### CODE REVIEW

**Comment by 6-6-6** [comment]
Looks good to me.

**Comment by leycec** [comment]
Thanks so much for the rapid merge, @6-6-6! Spark-overlay 4 life. :triumph:

File: dev-libs/apache-arrow/apache-arrow-9.0.0-r2.ebuild [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
 
-DESCRIPTION="A cross-language development platform for in-memory data."
+DESCRIPTION="A cross-language development platform for in-memory data"
 HOMEPAGE="https://arrow.apache.org/"
 SRC_URI="https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${PV}/${P}.tar.gz"
 
 LICENSE="Apache-2.0"
 SLOT="0"
-KEYWORDS="~amd64 ~x86"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
```
Comment (Leo3418) [inline dev-libs/apache-arrow/apache-arrow-9.0.0-r2.ebuild:19]: Please note that these keywords must not be added due to dependencies `dev-cpp/xsimd` and `dev-libs/thrift` not keyworded on those platforms.

Comment (heroxbd) [reply dev-libs/apache-arrow/apache-arrow-9.0.0-r2.ebuild:19]: Agreed.

**Pull request merged**